base/essentials.jl

```diff
@@ -467,10 +467,20 @@ Evaluate an expression with values interpolated into it using `eval`.
 If two arguments are provided, the first is the module to evaluate in.
 """
 macro eval(ex)
-    return Expr(:escape, Expr(:call, GlobalRef(Core, :eval), __module__, Expr(:quote, ex)))
+    g = ccall(:jl_gensym, Ref{Symbol}, ())
```

Comment:
I don't see why gensym should be needed?
```suggestion
    g = :eval_local_result
```
I am also not certain about the change in scope here

Reply:
> I am also not certain about the wisdom of the change in scope here either

I'm open to reconsidering that. It seemed like a practical tradeoff, considering you're likely to want to see the effect of it after, but if you think it's too confusing that it differs from the call version, we can drop it.

Reply:
I am mostly just worried it would be spuriously breaking (and thus surprising) for unsuspecting packages, which may or may not be part of PkgEval. It is such a subtle and unlikely thing, I am not even certain it can be noticed
```
julia> @eval begin $(x = 1) end # what is the scope of `x`?
```

Reply:
Fair enough. I'll use your suggestion below.

Reply:
Actually, I remember why I used `let` here. Since it's on the assignment line of the let, it gets evaluated in the outer scope. Only the latestworld and return get evaluated inside the let, but yes, the gensym is unnecessary.